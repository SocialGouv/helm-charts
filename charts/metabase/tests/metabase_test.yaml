suite: test maildev
#templates:
tests:
  - it: should render
    asserts:
      - template: deployment.yaml
        isKind:
          of: Deployment
      - template: deployment.yaml
        hasDocuments:
          count: 1
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: deployment.yaml
        matchSnapshot:
          path: spec
  - it: custom resources
    values:
      - values-resources.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec.containers[0].resources
  - it: custom tag
    values:
      - values-tag.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec.containers[0].image
