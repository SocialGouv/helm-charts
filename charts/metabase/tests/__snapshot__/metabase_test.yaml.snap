custom resources:
  1: |
    limits:
      cpu: 42m
      memory: 100m
    requests:
      cpu: 12m
      memory: 10m
custom tag:
  1: |
    metabase:42.12.1
should render:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: metabase
    template:
      metadata:
        labels:
          component: metabase
      spec:
        containers:
          - env: null
            envFrom: null
            image: metabase/metabase:v0.46.6.1
            livenessProbe:
              failureThreshold: 6
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 5
              timeoutSeconds: 5
            name: metabase
            ports:
              - containerPort: 3000
                name: http
            readinessProbe:
              failureThreshold: 15
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 1
              periodSeconds: 5
              successThreshold: 1
              timeoutSeconds: 1
            resources:
              limits:
                cpu: 1000m
                memory: 2048Mi
              requests:
                cpu: 500m
                memory: 512Mi
            securityContext:
              allowPrivilegeEscalation: false
            startupProbe:
              failureThreshold: 30
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 10
        securityContext:
          fsGroup: 1000
          runAsGroup: 1000
          runAsNonRoot: true
          runAsUser: 1000
  2: |
    replicas: 1
    selector:
      matchLabels:
        component: metabase
    template:
      metadata:
        labels:
          component: metabase
      spec:
        containers:
          - env: null
            envFrom: null
            image: metabase/metabase:v0.46.6.1
            livenessProbe:
              failureThreshold: 6
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 5
              timeoutSeconds: 5
            name: metabase
            ports:
              - containerPort: 3000
                name: http
            readinessProbe:
              failureThreshold: 15
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 1
              periodSeconds: 5
              successThreshold: 1
              timeoutSeconds: 1
            resources:
              limits:
                cpu: 1000m
                memory: 2048Mi
              requests:
                cpu: 500m
                memory: 512Mi
            securityContext:
              allowPrivilegeEscalation: false
            startupProbe:
              failureThreshold: 30
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 10
        securityContext:
          fsGroup: 1000
          runAsGroup: 1000
          runAsNonRoot: true
          runAsUser: 1000
  3: |
    replicas: 1
    selector:
      matchLabels:
        component: metabase
    template:
      metadata:
        labels:
          component: metabase
      spec:
        containers:
          - env: null
            envFrom: null
            image: metabase/metabase:v0.46.6.1
            livenessProbe:
              failureThreshold: 6
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 5
              timeoutSeconds: 5
            name: metabase
            ports:
              - containerPort: 3000
                name: http
            readinessProbe:
              failureThreshold: 15
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 1
              periodSeconds: 5
              successThreshold: 1
              timeoutSeconds: 1
            resources:
              limits:
                cpu: 1000m
                memory: 2048Mi
              requests:
                cpu: 500m
                memory: 512Mi
            securityContext:
              allowPrivilegeEscalation: false
            startupProbe:
              failureThreshold: 30
              httpGet:
                path: /api/health
                port: http
              initialDelaySeconds: 30
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 10
        securityContext:
          fsGroup: 1000
          runAsGroup: 1000
          runAsNonRoot: true
          runAsUser: 1000
